src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.artipie;
+
+import com.artipie.composer.http.PhpComposer;
+import com.artipie.files.FilesSlice;
+import com.artipie.gem.GemSlice;
+import com.artipie.http.Response;
+import com.artipie.http.Slice;
+import com.artipie.maven.http.MavenSlice;
+import com.artipie.npm.Npm;
+import com.artipie.npm.http.NpmSlice;
+import com.artipie.rpm.http.RpmSlice;
+import com.jcabi.log.Logger;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import org.reactivestreams.Publisher;
+
+/**
+ * Slice from repo config.
+ * @since 0.1.4
+ * @todo #90:30min We still don't have tests for Pie. But now that this class was extracted, we have
+ *  a more cohesive class that could be tested. Write unit tests for SliceFromConfig class.
+ */
+public final class SliceFromConfig implements Slice {
```

Comment:
@HDouss I don't think that this class can implement `Slice` since it has to build `CompletionStage<Slice>` in order to be async.

Reply:
@Vatavuk That's what the puzzle (issue description) tells. Anyway, I defered the slice building and response acting to the response method so it would be async.

Reply:
@HDouss it is not deferred because of instantiation of `php` slice which needs to fetch path from the repo. The code for `php` instantiation executes `.toCompletableFuture().get()`.

Reply:
@Vatavuk That still not happen until `SliceFromConfig.response` is called, which is deferred.